excellent/tests.go

```diff
@@ -392,6 +392,44 @@ func HasBeginning(env utils.Environment, args ...interface{}) interface{} {
 	return XFalseResult
 }
 
+// HasPattern tests whether `string` matches the regex `pattern`
+//
+// Both strings are trimmed of surrounding whitespace, but otherwise matching is strict
+// without any tokenization.
+//
+//   @(has_pattern("This is my Kazoo", "^kazoo")) -> false
+//   @(has_pattern("Kazoos are great", "^kazoo")) -> true
+//   @(has_pattern("Kazoos are great", "^kazoo").match) -> "Kazoo"
+//
+// @test has_pattern(string, pattern)
+func HasPattern(env utils.Environment, args ...interface{}) interface{} {
```

Comment:
Will need to take care of python regex differences: https://github.com/nyaruka/goflow/issues/73